### PR TITLE
Feature/ionic scaffold

### PR DESCRIPTION
**Pull request by emclaug2**
For PXBLUE-2063

#### Changes proposed in this Pull Request:

- Re-scaffold the ionic demo using the latest @pxblue/cli (1.7) and move the guts of the demo over

### CODE REVIEW

File: config.xml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 <?xml version='1.0' encoding='utf-8'?>
-<widget id="io.ionic.starter" version="0.0.1" xmlns="http://www.w3.org/ns/widgets" xmlns:cdv="http://cordova.apache.org/ns/1.0">+
-                                                                                                                                                                                                                                                                                          <name>ionic-design-patterns</name>
-    <description>Reusable design patterns for ionic.</description>
-    <author email="pxblue@eaton.com" href="https://pxblue.github.io">PX Blue</author>
+<widget id="io.ionic.starter" version="0.0.1" xmlns="http://www.w3.org/ns/widgets" xmlns:cdv="http://cordova.apache.org/ns/1.0">
+    <name>MyApp</name>
+    <description>An awesome Ionic/Cordova app.</description>
+    <author email="hi@ionicframework.com" href="http://ionicframework.com/">Ionic Framework Team</author>
```
Comment (joebochill) [inline config.xml:5]: This is one of the parts I commented on in Andrew's PR. This file gets auto generated I think, but I believe you can come in here and update the name, description, and author after it's created and they will be preserved.

File: ionic.config.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "ionic-design-patterns",
+  "name": "test",
```
Comment (joebochill) [inline ionic.config.json:2]: Did you name the project 'test'? Hopefully this is the only place that needs to be updated...that'll be a tricky string to find all of the instances of if it's elsewhere.

File: package.json [hunk 3]
```diff
@@ -1,103 +1,101 @@
...
+        "plugins": {
+            "cordova-plugin-whitelist": {},
+            "cordova-plugin-statusbar": {},
+            "cordova-plugin-device": {},
+            "cordova-plugin-splashscreen": {},
+            "cordova-plugin-ionic-webview": {},
+            "cordova-plugin-ionic-keyboard": {}
+        },
+        "platforms": [
+            "ios"
```
Comment (joebochill) [inline package.json:98]: You'll probably want to get the android platform in here too (you can clone your branch down on your pc and do the android bit)

**Review by joebochill — COMMENTED** [review]
Woohoo! The elusive green checkmark.


You need to delete these three lines in the config.xml:
image

They're some sort of holdover from an older version of cordova-res and they cause a build error (jeff and I came across this last week):

image

- The appbar scrolls of the page on most pages...not sure if you want to fix that now or log an issue for it (either is fine with me):
image

File: package.json (hunk 3 above)
Comment (emclaug2) [reply package.json:98]: I've just added the android platform but didnt push it out yet.

File: config.xml (hunk 1 above)
Comment (emclaug2) [reply config.xml]: Ive added the description and author here.

File: ionic.config.json (hunk 2 above)
Comment (emclaug2) [reply ionic.config.json]: You're right, I named the first app 'test' but I think I took care of all the instances of where it's used.

File: config.xml [hunk 4]
```diff
@@ -1,8 +1,8 @@
 <?xml version='1.0' encoding='utf-8'?>
 <widget id="io.ionic.starter" version="0.0.1" xmlns="http://www.w3.org/ns/widgets" xmlns:cdv="http://cordova.apache.org/ns/1.0">
-    <name>MyApp</name>
-    <description>An awesome Ionic/Cordova app.</description>
-    <author email="hi@ionicframework.com" href="http://ionicframework.com/">Ionic Framework Team</author>
+    <name>MyApp</name>                                                                                                                                                                                                  <name>ionic-design-patterns</name>
```
Comment (joebochill) [inline config.xml:3]: You've got two name tags here now

File: package.json [hunk 5]
```diff
@@ -98,4 +98,4 @@
             "ios"
         ]
     }
-}
+}
```
Comment (joebochill) [inline package.json:101]: I'm still not seeing android in here

File: config.xml (hunk 4 above)
Comment (emclaug2) [reply config.xml:3]: Good catch, I didn't see the one scrolled all the way to the right.

File: package.json (hunk 5 above)
Comment (emclaug2) [reply package.json:102]: Platform android has been added now

File: config.xml [hunk 6]
```diff
@@ -1,6 +1,6 @@
 <?xml version='1.0' encoding='utf-8'?>
 <widget id="io.ionic.starter" version="0.0.1" xmlns="http://www.w3.org/ns/widgets" xmlns:cdv="http://cordova.apache.org/ns/1.0">
-    <name>MyApp</name>                                                                                                                                                                                                  <name>ionic-design-patterns</name>
+    <name>MyApp</name>
```
Comment (joebochill) [inline config.xml:3]: You removed the more descriptive name tag in favor of the generic one?

**Review by joebochill — APPROVED** [review]

**Comment by jeffvg** [comment]
I jumped right in with iOS and had no issues. Android is giving me issues with this..

image

Then I realized I'm not able to run yarn start or ionic serv either

image

**Comment by joebochill** [comment]
@jeffvg do you by chance happen to have a package-lock.json and a yarn.lock file in your workspace? Ionic defaults to installing dependencies using npm unless you configure it to use yarn. If you have some things installed with npm and some with yarn, I've seen a message like that before.

**Comment by joebochill** [comment]
I just tried it and can confirm that is what happened for me (same error as you):
image

I deleted package-lock.json, reran yarn install, then it ran successfully.

image

**Comment by jeffvg** [comment]
> @jeffvg do you by chance happen to have a package-lock.json and a yarn.lock file in your workspace? Ionic defaults to installing dependencies using npm unless you configure it to use yarn. If you have some things installed with npm and some with yarn, I've seen a message like that before.

yes, I sure do. I have notes on this from a year ago...
* ionic config set yarn true
* ionic config set npmClient yarn
* ng config cli.packageManager yarn

**Review by jeffvg — APPROVED** [review]
after deleting package lock and running
ionic config set yarn true
ionic config set npmClient yarn
worked for me.
looks good on ios/android

**Comment by andrupu** [comment]
Hey @jeffvg, great teamwork on this:
image
We got this issue fixed.  We are good to go now.

**Review by daileytj — APPROVED** [review]

**Pull request merged**